Sources/ApolloCodegenLib/ApolloCodegen.swift

```diff
@@ -44,11 +60,27 @@ public class ApolloCodegen {
     guard graphqlErrors.isEmpty else {
       let errorlines = graphqlErrors.flatMap({ $0.logLines })
       CodegenLogger.log(String(describing: errorlines), logLevel: .error)
-      throw Error.validationFailed(atLines: errorlines)
+      throw Error.graphQLSourceValidationFailed(atLines: errorlines)
     }
 
     return try frontend.compile(schema: graphqlSchema, document: mergedDocument)
   }
+
+  static func generateSchemaFiles(
```

Comment:
This function right now is better suited to an integration test rather than a unit test; arguably the fact that this function doesn't return anything also goes against it being a static function. 🤔 

A better way might be to have separate functions that return the generators for each referenced type and `build(with:)` simply gathers and calls the generate function on them sequentially. At least that would be testable to ensure `ApolloCodegen` is creating the correct generators from a given schema. That also means the file generators would need to hold on to values (referenced type and path) so their generate function doesn't need parameters.

Reply:
I have no objections to making this not static. I think your way sounds reasonable.